src/config.ts

```diff
@@ -30,12 +39,18 @@ const readConfig = (): Config => {
 };
 
 export const setConfigDir = (directory: string): void => {
-  const configFile = resolve(directory, CONFIG_FILE_NAME);
-  if (!fs.existsSync(configFile)) {
-    throw new NotFoundError(resolve(configDir));
+  const foundFile = [CONFIG_FILE_NAME, CONFIG_FILE_NAME_TS].find((config) => {
+    const file = resolve(directory, config);
+    return fs.existsSync(file);
+  });
+
+  if (!foundFile) {
+    throw new NotFoundError(resolve(directory));
   }
 
+  process.env.JEST_DYNALITE_CONFIG_FILE = foundFile;
   process.env.JEST_DYNALITE_CONFIG_DIRECTORY = directory;
+  configFile = foundFile;
```

Comment:
Higher up we have `let configFile: string = process.env.JEST_DYNALITE_CONFIG_FILE || CONFIG_FILE_NAME;` which is great, but then here where you set foundFile, you are using `[CONFIG_FILE_NAME, CONFIG_FILE_NAME_TS].find`. I think this logic could be improved, as it is depending on setConfigDIr and readConfig to be called in the correct order - relying on side effects. Also setting `process.env.JEST_DYNALITE_CONFIG_FILE = foundFile;` and `configFile = foundFile;` feels like it could be avoided.

Reply:
You are right, this is complex. I've put in stuff to simplify